src/plugins/intel_cpu/src/nodes/common/permute_kernel.cpp

```diff
@@ -220,6 +220,15 @@ void PermuteKernel::optimizedExecute(const uint8_t* src_data, uint8_t* dst_data,
     }
 
     switch (jcp.n) {
+    case 0:
+    // This case will trigger regression if the transpose connect from Parameter to Result
+    //  Parameter
+    //     |
+    //  Transpose
+    //     |
+    //  Result
+    // The elimination of the Transpose node will not be performed
+    // So copy from input buffer to output buffer without any permutation
```

Comment:
If n == 0 then the input, output should be scalar values (single value), then it should be enough to just copy the value
```suggestion
    // This case will trigger regression if the transpose connect from Parameter to Result
    //  Parameter
    //     |
    //  Transpose
    //     |
    //  Result
    // The elimination of the Transpose node will not be performed
    // So copy from input buffer to output buffer without any permutation
    *dst_data = *src_data;
    break;
```

Reply:
The parameter `n` looks more like the amount of permute axis. So, if `n == 0` it doesn't mean that this is a scalar, it can very well be a 1 rank tensor.
https://github.com/openvinotoolkit/openvino/blob/382a1ee4ddc4f994a7f0946a30a9b48473aa3fe9/src/plugins/intel_cpu/src/nodes/executors/transpose.cpp#L115-L130
In fact, the test represents rather a degenerate use case, and in real networks, there are more than one such layer, therefore such a Transpose is optimized out.